### ISSUE TITLE
Problems encountered in Redis data persistence and data recovery, urgent request for response!!!

### ISSUE THREAD

**Issue by nb001** [RediSearch/RediSearch#3744]
If Redis is started from a Docker container, the Redis Redisearch module cannot be started simultaneously during container startup. It needs to be loaded into the container after startup using the command MODULE LOAD /usr/lib/redis/modules/redisearch.so.

When performing data persistence, suppose a new container is started using the following command, and the container mounts the data in the /home/redis/data4 folder on the host:

docker run -p 6379:6379 --name redisnewdata3
-v /home/redis/data4:/data
-v /home/redis/conf/redis.conf:/etc/redis/redis.conf
-d docker.io/redislabs/redisearch
--appendonly yes --requirepass abc123456

If the /home/redis/data4 folder contains any Redisearch commands in the appendonly.aof file, such as "FT.CREATE myindex ON HASH SCHEMA reminder TEXT", the entire container will fail to start because the "redisearch" module cannot be loaded and the data in the "appendonly.aof" file cannot be recovered. Even if the container is started and the redisearch module is loaded first, replacing the data in the /home/redis/data4 folder will still result in the data not being loaded and recovered properly.

In other words, if the data file contains any commands associated with modules like Redisearch and RediJSON, Docker containers will not be able to recover the data from the 'appendonly.aof' file because the modules cannot be loaded during container startup. This inability to load the modules will result in the container failing to start correctly, causing a deadlock. The container startup logs will display the following error: Unknown command 'FT._CREATEIFNX' reading the append only file.

Similarly, if the data recovery is changed to an RDB file, the container startup will also fail to start correctly because it cannot recognize the AUX.

I am eagerly awaiting your response and hope that you continue to improve.

**Comment by oshadmi** [comment]
@nb001 You need to use a container which has the required modules installed in order to be able to load module data.
Please consider using [redis-stack](https://redis.io/docs/getting-started/install-stack/docker/)
